cdhweb/static_src/components/MainNavDesktop/MainNavDesktopItem.tsx

```diff
@@ -0,0 +1,161 @@
+import { MouseEvent, useId } from 'react';
+import Svg from '../svg/svg';
+import cx from 'classnames';
+import { MainNavItemPrimary, NavItem } from '../../data-types';
+
+type DesktopMenuItemProps = {
+  item: MainNavItemPrimary;
+  isOpen: boolean;
+  onClick: (e: MouseEvent<HTMLElement>) => void;
+  dropdownRef: React.RefObject<HTMLDivElement>;
+};
+
+type SubmenuItemProps = {
+  item: NavItem;
+  isLast?: boolean;
+};
+
+const SubmenuItem = ({ item, isLast }: SubmenuItemProps) => (
+  <a
+    href={item.link_url}
+    className="main-nav-desktop-submenu-link"
+    data-is-last={isLast || undefined}
+  >
+    {item.title}
+  </a>
+);
+
+type SubmenuColumnsProps = {
+  items: NavItem[];
+};
+
+// Split list of links into two columns,but only if the number of links meets a `threshold`.
+// Yes we can do this in CSS but it feels a bit nicer if the reading order goes "top to bottom,
+// left to right" rather than zig-zagging "left to right, top to bottom".
```

Comment:
you could do it like this: https://codepen.io/cameralibre/pen/vYwNEqg
(But this is a JS component anyway so there's not much point in ✨CSS-only✨ acrobatics)

Reply:
Ah, you've reminded me that I prototyped this thing a while back:
https://codepen.io/liamj/pen/oNwYEpo?editors=1100
It shows the finished DOM, but not how to get there. But I guess It'd be a matter of applying that `1` or `2` in the inline style based on similar JS maths, but we'd end up with a single container rather than 2. Sl, slightly nicer maybe, but probably also not worth the refactor at this stage.

Reply:
oops, just realised that the CSS was broken in my pen before 😓

Reply:
ah yes, that's better. that's nice in that it doesn't touch the DOM. but having to add lots of `:nth-child(x)` isn't ideal either.

actually wait, I think I could just do something with `:nth-child(odd/even)`... surely that'd be too obvious/easy 🤔 and su rely i already tried that originally. hmm.

Reply:
oh yeah, duh, that won't work. the whole thing is needing to know how many items there are, not which are odd/even (which is the default layout)